reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/TakeObservable.kt

```diff
@@ -0,0 +1,43 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+
+/**
+ * Emit only the first [limit] items emitted by source.
+ */
+fun <T> Observable<T>.take(limit: Int): Observable<T?> {
+    require(limit > 0) { "count >= 0 required but it was $limit" }
+
+    return observable { emitter ->
+        val disposableWrapper = DisposableWrapper()
+        emitter.setDisposable(disposableWrapper)
+
+        var remaining = limit
+
+        subscribeSafe(object : ObservableObserver<T?> {
```

Comment:
```kotlin
object : ObservableObserver<T>, ErrorCallback by emitter, CompleteCallback by emitter {
    fun onNext()
    fun onSubscribe()
}
```

Reply:
Just `CompletableCallbacks by emitter` will work as well. `CompletableCallbacks` extends both `ErrorCallback` and `CompleteCallback`.